src/main/java/io/cdap/plugin/dlp/SensitiveRecordFilter.java

```diff
@@ -0,0 +1,319 @@
+
+  private static final Logger LOG = LoggerFactory.getLogger(SensitiveRecordFilter.class);
+  public static final String NAME = "SensitiveRecordFilter";
+  public static final String DESCRIPTION = "Filters input records based that are sensitive.";
+  private static final String SENSITIVE_PORT = "Sensitive";
+  private static final String NON_SENSITIVE_PORT = "Non-Sensitive";
+
+  // Stores the configuration passed to this class from user.
+  private final Config config;
+
+  private StageMetrics metrics;
+
+  // DLP service client for managing interactions with DLP service.
+  private DlpServiceClient client;
+
+  @VisibleForTesting
+  public SensitiveRecordFilter(Config config) {
+    this.config = config;
+  }
+
+  /**
+   * Invoked during deployment of pipeline to validate configuration of the pipeline. This method checks if the input
+   * specified is 'field' type and if it is, then checks if the field specified is present in the input schema.
+   *
+   * @param configurer a <code>MultiOutputPipelineConfigurer</code> for configuring pipeline.
+   * @throws IllegalArgumentException if there any issues with configuration of the plugin.
+   */
+  @Override
+  public void configurePipeline(MultiOutputPipelineConfigurer configurer) {
+    super.configurePipeline(configurer);
+
+    MultiOutputStageConfigurer stageConfigurer = configurer.getMultiOutputStageConfigurer();
+    Schema inputSchema = stageConfigurer.getInputSchema();
+
+    if (client == null) {
+      try {
+        LOG.error(getSettings().toString());
+        client = DlpServiceClient.create(getSettings());
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+
+    config.validate(stageConfigurer.getFailureCollector(), inputSchema);
+
+    Map<String, Schema> outputs = new HashMap<>();
+    outputs.put(SENSITIVE_PORT, inputSchema);
+    outputs.put(NON_SENSITIVE_PORT, inputSchema);
+    stageConfigurer.setOutputSchemas(outputs);
+  }
+
+  /**
+   * Initialize this <code>SensitiveRecordFilter</code> plugin. A instance of DLP client is created with mapped
+   * infotypes.
+   *
+   * @param context Initialization context
+   */
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    super.initialize(context);
+
+    client = DlpServiceClient.create(getSettings());
+    metrics = context.getMetrics();
+  }
+
+  @Override
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    super.prepareRun(context);
+
+    String templateName = String.format("projects/%s/inspectTemplates/%s", config.getProject(), config.templateId);
+    GetInspectTemplateRequest request = GetInspectTemplateRequest.newBuilder().setName(templateName).build();
+
+    try {
+      InspectTemplate template = client.getInspectTemplate(request);
+
+    } catch (Exception e) {
+      throw new IllegalArgumentException(
+        "Unable to validate template name. Ensure template ID matches the specified ID in DLP");
```

Comment:
should also happen while doing validation

Reply:
Doing this check during validation is difficult since the request needs to access the DLP instance in the customer project and validation is run from the tenant project

Reply:
how is it different than doing validation for BigQuery plugins?